app/services/report_importer.rb

```diff
@@ -2,14 +2,38 @@ class ReportImporter
   delegate :logger, :to => :Rails
   attr_reader :report
 
+  def self.importer_for(type)
+    importers[type.to_s] || importers[:puppet]
+  end
+
+  def self.importers
+    @importers ||= { :puppet => ConfigReportImporter }.with_indifferent_access
+  end
+
+  def self.register_report_importer(key, klass)
+    importers[key.to_sym] = klass
+  end
+
+  def self.fact_features
+    importers.map { |_type, importer| importer.authorized_smart_proxy_features }.compact.flatten.uniq
+  end
+
+  def self.authorized_smart_proxy_features
+    Rails.logger.debug("Importer #{self} does not implement authorized_smart_proxy_features.")
+    Foreman::Deprecation.deprecation_warning('1.12', "ReportImporter is a superclass, please use child classes. Maybe try: ConfigReportImporter?")
```

Comment:
I don't understand the deprecation warnings here - they're going to fire all the time with the ReportsController and I don't see what replaces them.  The controller is always going to call ReportImporter.authorized_smart_proxy_features, not some Report or ReportImporter subclass.

They're also for a new method, which doesn't make a lot of sense.